File: language/runtime/android/kotlin/src/main/kotlin/dev/destack/runtime/android/input/text/TextInputEvents.kt
package dev.destack.runtime.android.input.text

/**
 * The text-input ingress surface attached to one Android runtime host.
 */
public fun interface TextInputEvents {
    /**
     * Send one host text-input event into the attached runtime session.
     */
    public fun sendTextInputEvent(
        event: RuntimeHostTextInputEvent,
    )
}

File: language/runtime/apple/core/Sources/RuntimeHostAppleCore/Input/Text/TextInputEvents.swift
import Foundation

/// The text-input ingress surface attached to one Apple runtime host.
@MainActor
public protocol TextInputEvents: AnyObject {
  /// Send one host text-input event into the attached runtime session.
  func sendTextInputEvent(_ event: RuntimeHostTextInputEvent)
}

/// The explicit no-op text-input ingress surface for one Apple runtime host.
@MainActor
public final class NoopTextInputEvents: TextInputEvents {
  /// Create one no-op text-input ingress surface.
  public init() {}

  public func sendTextInputEvent(_ event: RuntimeHostTextInputEvent) {
    let _ = event
  }
}
